synology_dsm/api/core/share.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+"""Shared Folders data."""
+from synology_dsm.helpers import SynoFormatHelper
+
+
+class SynoShare(object):
+    """Class containing Share data."""
+
+    API_KEY = "SYNO.Core.Share"
+    # Syno supports two methods to retrieve resource details, GET and POST.
+    # GET returns a limited set of keys. With POST the same keys as GET
+    # are returned plus any keys listed in the "additional" parameter.
+    # NOTE: The value of the additional key must be a string.
+    REQUEST_DATA = {
+        "additional": '["hidden","encryption","is_aclmode","unite_permission","is_support_acl",'
+        '"is_sync_share","is_force_readonly","force_readonly_reason","recyclebin",'
+        '"is_share_moving","is_cluster_share","is_exfat_share","is_cold_storage_share",'
+        '"support_snapshot","share_quota","enable_share_compress","enable_share_cow",'
+        '"include_cold_storage_share","is_cold_storage_share"]',
+        "shareType": "all",
+    }
+
+    def __init__(self, dsm):
+        self._dsm = dsm
+        self._data = {}
+
+    def update(self):
+        """Updates share data."""
+        raw_data = self._dsm.post(self.API_KEY, "list", data=self.REQUEST_DATA)
+        if raw_data:
+            self._data = raw_data["data"]
+
+    @property
+    def shares(self):
+        """Gets all shares."""
+        return self._data.get("shares", [])
+
+    def share(self, share_name):
```

Comment:
```suggestion
    def get_share(self, share_id):
```

As in storage.

We can make `storage._get_disk` and  `storage._get_volume` public too.

Reply:
Done, with a note: Syno uses `uuid`, not `id` for shares so I'm using it as well - 
```
def get_share(self, share_uuid):
```